lock.go

```diff
@@ -134,9 +151,30 @@ func (l *Lock) LockWithData(data []byte) error {
 			continue
 		}
 
-		ev := <-ch
-		if ev.Err != nil {
-			return ev.Err
+		if timeout >= 0 {
```

Comment:
This is a timeout waiting for events of acquiring a lock. It does not actually bound the lock timing at all outside of how long we wait. If this pattern of Trying to lock within a time boundary it would also bound the creating of znodes as well, or be very clearly documented that this duration is only applied on how long we watch for events on failure to acquire a lock.